### PR TITLE
Enhance Ethereum compatibility while deploying contracts

### PR DESCRIPTION
**Pull request by asoto-iov**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size [hunk 1]
```diff
@@ -0,0 +1,66 @@
+
+comment
+
+// Contracts compiled using 0.8.20
+// the contract source code
+
+contract BytecodeFactory {
+    event ContractDeployed(address contractAddress);
```
Comment (fmacleal) [inline rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size:8]: question:

I see that you are emitting this event but not verifying it in the DSL test. You could check the emitted event like [this](https://github.com/rsksmart/rskj/blob/314708cf08c792080ba02c90015e3ecd477cccad/rskj-core/src/test/java/co/rsk/vm/opcode/TransientStorageDslTest.java#L288). Or, just delete the emit event contract part. It doesn't seem to bring value.

**Review by fmacleal — APPROVED** [review]
Good job. I see that the coverage seems very good with the contracts.
I just have a few comments with simple suggestions, but nothing that prevent the approval.

File: rskj-core/src/test/java/org/ethereum/core/ContractCreatingRollbackTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import co.rsk.test.dsl.DslProcessorException;
+import co.rsk.test.dsl.WorldDslProcessor;
+import com.typesafe.config.ConfigValueFactory;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileNotFoundException;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ContractCreatingRollbackTest {
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/test/java/org/ethereum/core/ContractCreatingRollbackTest.java]: Addressed, renamed the class to `ContractCreatingDslRollbackTest`

File: rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+    // Función para desplegar un contrato con `create2` a partir de un bytecode conocido
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size]: Nice suggestion, renamed it to `deployContractWithCreate2WithTheCodePassedInParameter`

File: rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    function deployContract(bytes memory bytecode, uint256 salt) public returns (address) {
+        address contractAddress;
+
+        assembly {
+            // Crear el contrato usando `create2` y asignar la dirección a `contractAddress`
```
[the remark this answers is not in the document]
Comment (rmoreliovlabs) [reply rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size]: Addressed

File: rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size (hunk 1 above)
Comment (rmoreliovlabs) [reply rskj-core/src/test/resources/dsl/453_rollback_during_creation/exceeding_code_size]: Removed the emit event for the contract, as it’s not being used or checked. :)

**Review by Vovchyk — APPROVED** [review]

**Review by fmacleal — DISMISSED** [review]
Good job!
Thanks for have addressed the comments. :)

**Review by Vovchyk — APPROVED** [review]

**Review by nagarev — APPROVED** [review]
LGTM! :shipit:

**Review by fmacleal — APPROVED** [review]
LGTM

**Pull request merged**